### PR TITLE
Bugfix solve 80_optimization

### PR DESCRIPTION
**Pull request by flohump**
## :bird: Description of this PR :bird:

- **80_optimization** fixed a bug in nlp_apr17; cycling through CONOPT4, CONOPT4 without preprocessing and CONOPT3 was not working
- **58_peatland** Added balance variable to avoid random infeasibilites

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

**Review by tscheypidi — APPROVED** [review]

File: start.R [hunk 1]
```diff
@@ -97,13 +97,13 @@ runOutputs <- function(runscripts=NULL, submit=NULL) {
         log <- format(Sys.time(), paste0("logs/", rout_name, "-%Y-%H-%M-%S-%OS3.log"))
         system2("Rscript",name, stderr = log, stdout = log, wait=FALSE)
       } else if(submit=="slurmpriority") {
-        system(paste(sbatch_command,"--qos=priority"))
+        system(paste(sbatch_command,"--qos=priority --cpus-per-task=3"))
```
Comment (pascal-sauer) [inline start.R:100]: What is the reason for this change? Does magpie need more resources now?

**Review by pascal-sauer — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - **scenario_config.csv** same revision for input files as in default.cfg
 - **scenario_fsec.csv** scenario settings
 - **start/projects/fsec.R** scenario settings
+- **80_optimization** fixed a bug in nlp_apr17; cycling through CONOPT4, CONOPT4 without preprocessing and CONOPT3 was not working
```
[the remark this answers is not in the document]
Comment (flohump) [reply CHANGELOG.md:25]: It's happening in solve.gms here:
https://github.com/magpiemodel/magpie/pull/666/files#diff-646b34de332057b04db43a5de3ae3f570a597172b819976733676f74c05bb519L49
Purpose: in case of problems try different solvers and optfile settings
The loop was not really working because `magpie.optfile` was overwritten by the default setting before saving it to `s80_optfile_previter`

File: start.R (hunk 1 above)
Comment (flohump) [reply start.R:100]: If you have a start script that starts a MAgPIE run with sequential = FALSE in the submission job, memory issues are likely to occur because only 1 CPU is used by default. Therefore I increased this now to 3 CPUs, which solves the problem.

**Review by pascal-sauer — APPROVED** [review]

**Review by pascal-sauer — APPROVED** [review]
